latch_cli/services/cp/utils.py

```diff
@@ -0,0 +1,199 @@
+import os
+import re
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import Path
+from typing import Dict, TypedDict
+from urllib.parse import urlparse
+
+import gql
+
+from latch_cli.config.user import user_config
+from latch_cli.services.cp.exceptions import AuthenticationError, PathResolutionError
+
+
+def get_auth_header() -> Dict[str, str]:
+    sdk_token = user_config.token
+    execution_token = os.environ.get("FLYTE_INTERNAL_EXECUTION_ID")
+
+    if sdk_token is not None and sdk_token != "":
+        headers = {"Authorization": f"Latch-SDK-Token {sdk_token}"}
+    elif execution_token is not None:
+        headers = {"Authorization": f"Latch-Execution-Token {execution_token}"}
+    else:
+        raise AuthenticationError("Unable to find authentication credentials.")
+
+    return headers
+
+
+is_valid_path_expr = re.compile(r"^(latch)?://")
```

Comment:
This regex matches `://dsjd` as well as `latch://dsjd` -- you want `^latch://`

Reply:
this is intentional, max wanted the scheme to be optional